un-SCL-ify hammer-cli-katello

ianballou opened the pull request:
Makes the hammer-cli-katello spec file no longer use ye olde SCL macros from the EL7 days.

evgeni commented on packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec:
```diff
@@ -1,77 +1,50 @@
 %global gem_name hammer_cli_katello
 %global plugin_name katello
 
-%global release 1
-%global prereleasesource pre.master
-%global prerelease %{?prereleasesource:.}%{?prereleasesource}
```
I think you want to keep *these* to allow it to be nightly built

ianballou replied:
Yeah I just realized that, I'll fix the naming.

ekohl commented on packages/katello/rubygem-hammer_cli_katello/hammer_cli_katello-1.12.0.gem:
```diff
@@ -0,0 +1 @@
+../../../.git/annex/objects/9J/ZK/SHA256E-s2427904--a680b8e9f05374d0410ad77214067881b4d95cabed95f2c74a71df2bba9d3c16.0.gem/SHA256E-s2427904--a680b8e9f05374d0410ad77214067881b4d95cabed95f2c74a71df2bba9d3c16.0.gem
```
For a nightly package we don't "store" the gem. It's retrieved from Jenkins.

ekohl commented on packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec:
```diff
@@ -1,77 +1,57 @@
+Requires: ruby(release)
+Requires: ruby
+Requires: ruby(rubygems)
+Requires: rubygem(hammer_cli_foreman)
+Requires: rubygem(hammer_cli_foreman_tasks)
```
We should look at the autorequires. This may be redundant but then we need to update the template too.

ekohl replied:
Looking again, I think you did this by hand. In 3f4c04273594a374a1f04b1ad0441c31a03ca237 we redid this section to rely on autorequires.

```suggestion
```

While we're on the topic of updates, in 10e0424442e2a8d9703d83e07801218676a480f6 we started to add Provides for plugins. Could you add that too?

ekohl commented on packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec:
```diff
@@ -88,14 +68,17 @@ install -m 0644 .%{gem_instdir}/config/%{plugin_name}.yml \
 %{gem_instdir}/test
 
 %changelog
+* Thu Apr 25 2024 Ian Ballou <ianballou67@gmail.com> 1.12.0-1
+- Update to 1.12.0-1
```
This should be a nightly version.

ekohl reviewed (commented):
I'd like to point out (once again) that since https://github.com/Katello/hammer-cli-katello/pull/924 the nightly packages have been broken: https://ci.theforeman.org/blue/organizations/jenkins/hammer-cli-katello-master-source-release/detail/hammer-cli-katello-master-source-release/204/pipeline

Given they've been broken for 3 months, can we assume they're not tested anyway and we can change it regular gems?

ianballou commented:
> Given they've been broken for 3 months, can we assume they're not tested anyway and we can change it regular gems?

There is a testing queue for hammer-cli-katello issues that has built up over the 3 months, but it's small enough for people not to be clamoring about it. We would like to keep the h-c-k nightlies going. Last I heard, Zach was taking a look.

ekohl commented:
You'll need to recreate Gemfile and define a testing group. Jenkins now doesn't install the development group

ianballou commented:
I added a test group to https://github.com/Katello/hammer-cli-katello/pull/934/files

ianballou commented:
Let's merge https://github.com/theforeman/foreman-packaging/pull/10767 first to unblock nightly if this PR still has issues.

evgeni commented on packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec:
```diff
@@ -1,77 +1,57 @@
-%{?scl:scl enable %{scl} - << \EOF}
-gem build %{gem_name}.gemspec
-%{?scl:EOF}
+gem build ../%{gem_name}-%{version}.gemspec
```
```suggestion
gem build ../%{gem_name}-%{version}%{?prerelease}.gemspec
```

I think that's the last place missing `prerelease`.

ekohl commented on packages/katello/rubygem-hammer_cli_katello/rubygem-hammer_cli_katello.spec:
```diff
@@ -1,77 +1,57 @@
+BuildRequires: rubygems-devel
 BuildArch: noarch
-Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: rubygem(%{gem_name}) = %{version}
```
I think this is automatic nowadays and was dropped in 3f4c04273594a374a1f04b1ad0441c31a03ca237

ianballou commented:
@ekohl to address the comments about items being different from what the spec automation provides, I rebased, regenerated the entire spec file, and added back in the nightly bits.

ianballou commented:
```
[2024-05-15T20:58:50.452Z] TASK [Check for a new changelog entry] *****************************************

[2024-05-15T20:58:50.452Z] fatal: [rubygem-hammer_cli_katello]: FAILED! => 

[2024-05-15T20:58:50.452Z]     changed: false

[2024-05-15T20:58:50.452Z]     changelog_epoch_version_release: ''

[2024-05-15T20:58:50.452Z]     msg: Could not find specfile

[2024-05-15T20:58:50.452Z]     specfile_epoch_version_release: ''

```

I'm unsure why the specfile cannot be found.

evgeni commented:
It added another hammer cli katello to plugins in comps/manifest, that confuses it.

evgeni commented:
I've dropped those additional entries and it built \o/

@ekohl mind having another look?
